src/type_helpers.rs

```diff
@@ -1,58 +1,57 @@
-use named_type::NamedType;
 use serde::{de::DeserializeOwned, ser::Serialize};
 use snafu::ResultExt;
 
 use cosmwasm::errors::{NotFound, ParseErr, Result, SerializeErr};
 use cosmwasm::serde::{from_slice, to_vec};
 
+pub fn short_type_name<T>() -> &'static str {
+    let long = std::any::type_name::<T>();
+    long.rsplit("::").next().unwrap_or(long)
+}
```

Comment:
We only need this in one file, right? Can we make it non-public?

We should not expoose this to types we don't control, as this can lead to undesired results. On the one hand it will shorten `Option<String>` to `String>` and on the other hand we don't even have guaranteed on the output of `type_name`:

> This is intended for diagnostic use. The exact contents and format of the string are not specified, other than being a best-effort description of the type. For example, type_name::<Option<String>>() could return the "Option<String>" or "std::option::Option<std::string::String>", but not "foobar". In addition, the output may change between versions of the compiler.

https://doc.rust-lang.org/std/any/fn.type_name.html#note

Reply:
Yes, it should be non public.

We can use the long name as well, but it's only intended for the structs we serialize

Reply:
As long as we know what kind of inputs we use, the method makes sense. Let's just make it non-`pub`.